File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    
    /*
    ex) 
    prices = [7,1,5,3,6,4]
                  
                
                
   min = 1
   max = max(prices[i] - min) = 4
    
    */
    
   let min = prices[0];
    let max = 0;
    for(let i=1; i<prices.length; i++){
        min = Math.min(min, prices[i]);
        max = Math.max(max, prices[i] - min)
    }
    return max
    
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left = 0
        maxProfit = 0
        for right in range(1, len(prices)):
            if prices[left] >= prices[right]:
                left = right
                
            
            else:
                maxProfit = max(maxProfit, prices[right]-prices[left])
                
        return maxProfit
        